File: medplum-java/medplum-java-model/src/main/java/com/fhir/InsurancePlan_Coverage.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableInsurancePlan_Coverage.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface InsurancePlan_Coverage {
  Optional<List<Extension>> modifierExtension();

  List<InsurancePlan_Benefit> benefit();

  CodeableConcept type();

  Optional<List<Reference>> network();

  Optional<List<Extension>> extension();

  Optional<String> id();

  static ImmutableInsurancePlan_Coverage.TypeBuildStage builder() {
    return ImmutableInsurancePlan_Coverage.builder();
  }
}

File: medplum-typescript/src/model/fhir/InsurancePlan_Coverage.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { InsurancePlan_Benefit } from "./InsurancePlan_Benefit";
import { Reference } from "./Reference";

export interface InsurancePlan_Coverage {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: CodeableConcept;
    network: Reference[] | null | undefined;
    benefit: InsurancePlan_Benefit[];
}
